tests/snapshot/test_snapshot.py

```diff
@@ -32,53 +27,54 @@ def test_snapshot_restore_and_drop_in_parallel(self):
-                # Drop snapshot2 while the restore of snapshot1 is still running
-                c.execute('RESTORE SNAPSHOT repo.snapshot1 ALL WITH (wait_for_completion = false)')
-                try:
-                    c.execute('DROP SNAPSHOT repo.snapshot2')
-                except ProgrammingError:
-                    self.fail("Restore and Drop Snapshot operation should work in parallel")
-
-                assert_busy(lambda: self._assert_num_docs(conn, num_docs))
-
-            cluster.stop()
+        try:
+            with MinioServer() as minio:
+                t = threading.Thread(target=minio.run)
+                t.daemon = True
+                t.start()
+                wait_until(lambda: _is_up('127.0.0.1', 9000))
+
+                num_nodes = random.randint(3, 5)
+                number_of_shards = random.randint(1, 3)
+                number_of_replicas = random.randint(0, 2)
+                num_docs = random.randint(1, 100)
+
+                cluster_settings = {
+                    'cluster.name': gen_id(),
+                }
+                cluster = self._new_cluster('branch:master', num_nodes, settings=cluster_settings)
+                cluster.start()
+
+                with connect(cluster.node().http_url, error_trace=True) as conn:
+                    c = conn.cursor()
+                    wait_for_active_shards(c)
+                    c.execute('''
+                                create table doc.test(x int) clustered into ? shards with( number_of_replicas =?)
+                             ''', (number_of_shards, number_of_replicas,))
+
+                    insert_data(conn, 'doc', 'test', num_docs)
+
+                    c.execute('''
+                                CREATE REPOSITORY repo TYPE S3
+                                WITH (access_key = 'minio',
+                                secret_key = 'miniostorage',
+                                bucket='backups',
+                                endpoint = '127.0.0.1:9000',
+                                protocol = 'http')
+                            ''')
+
+                    c.execute('CREATE SNAPSHOT repo.snapshot1 TABLE doc.test WITH (wait_for_completion = true)')
+                    c.execute('CREATE SNAPSHOT repo.snapshot2 TABLE doc.test WITH (wait_for_completion = true)')
+                    c.execute('DROP TABLE doc.test')
+                    # Drop snapshot2 while the restore of snapshot1 is still running
+                    c.execute('RESTORE SNAPSHOT repo.snapshot1 ALL WITH (wait_for_completion = false)')
+                    try:
+                        c.execute('DROP SNAPSHOT repo.snapshot2')
+                    except ProgrammingError:
+                        self.fail("Restore and Drop Snapshot operation should work in parallel")
+
+                    assert_busy(lambda: self._assert_num_docs(conn, num_docs))
+
+                cluster.stop()
+        finally:
+            self.tearDown()
```

Comment:
we used to use one specific folder  and then clean it up with `shutil.rmtree(self.PATH_DATA, ignore_errors=True)`
now each node has it's own folder which cleanup on teardown